frontend/src/metabase/query_builder/components/expressions/Editor/Editor.tsx

```diff
@@ -93,39 +92,51 @@ export function Editor<S extends StartRule = "expression">(
     stageIndex,
     name,
     expressionIndex,
-    onCommit,
     reportTimezone,
     metadata,
     extensions: [customTooltip],
-    shortcuts,
   });
 
-  const handleBlur = useCallback(() => {
-    onChange(source);
-  }, [source, onChange]);
-
   return (
-    <>
-      <div className={cx(S.wrapper, { [S.formatting]: isFormatting })}>
-        <div className={S.prefix}>=</div>
-        <CodeMirror
-          id={id}
-          ref={ref}
-          data-testid="custom-expression-query-editor"
-          className={S.editor}
-          extensions={extensions}
-          readOnly={readOnly || isFormatting}
-          value={source}
-          onChange={onSourceChange}
-          onBlur={handleBlur}
-          height="100%"
-          width="100%"
-          indentWithTab={false}
+    <Flex
+      className={cx(S.wrapper, { [S.formatting]: isFormatting })}
+      direction="column"
+    >
+      <CodeMirror
+        id={id}
+        ref={ref}
+        data-testid="custom-expression-query-editor"
+        className={S.editor}
+        extensions={extensions}
+        readOnly={readOnly || isFormatting}
+        value={source}
+        onChange={onSourceChange}
+        onBlur={onBlur}
+        height="100%"
+        width="100%"
+        indentWithTab={false}
+      />
+      <Errors error={source !== "" ? error : null} />
+
+      <Shortcuts
+        shortcuts={shortcuts}
+        hide={isFormatting || source.trim() !== "" || error != null}
+      />
+
+      <Box className={S.toolbar} p="xs">
+        <Button
+          title={t`Format`}
+          aria-label={t`Format`}
+          onClick={formatExpression}
+          variant="subtle"
+          leftSection={<Icon name="snippet" />}
+          size="xs"
+          disabled={Boolean(isFormatting || error)}
```

Comment:
1. Add custom column
2. Click "Format"

Shortcuts disappear.

I think the button should be initially disabled (or even hidden?).